crates/oxc_linter/src/rules/eslint/radix.rs

```diff
@@ -0,0 +1,228 @@
+                }
+            }
+            _ => {
+                let radix_arg = &call_expr.arguments[1];
+                if matches!(&self.radix_type, RadixType::AsNeeded) && is_default_radix(radix_arg) {
+                    ctx.diagnostic(RadixDiagnostic::RedundantRadix(radix_arg.span()));
+                } else if !is_valid_radix(radix_arg) {
+                    ctx.diagnostic(RadixDiagnostic::InvalidRadix(radix_arg.span()));
+                }
+            }
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+enum RadixType {
+    #[default]
+    Always,
+    AsNeeded,
+}
+
+impl RadixType {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "as-needed" => Self::AsNeeded,
+            _ => Self::Always,
+        }
+    }
+}
+
+fn is_default_radix(node: &Argument) -> bool {
+    node.to_expression().is_specific_raw_number_literal("10")
+}
+
+fn is_valid_radix(node: &Argument) -> bool {
+    let expr = node.to_expression();
+
+    if let Expression::NumericLiteral(lit) = expr {
+        return lit.value.fract() == 0.0 && lit.value >= 2.0 && lit.value <= 36.0;
+    }
+
+    if let Expression::Identifier(_) = expr {
+        return !expr.is_undefined();
+    }
+
+    false
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+    use serde_json::json;
+
+    let pass = vec![
+        (r#"parseInt("10", 10);"#, None),
+        (r#"parseInt("10", 2);"#, None),
+        (r#"parseInt("10", 36);"#, None),
+        (r#"parseInt("10", 0x10);"#, None),
+        (r#"parseInt("10", 1.6e1);"#, None),
+        (r#"parseInt("10", 10.0);"#, None),
+        (r#"parseInt("10", foo);"#, None),
+        (r#"Number.parseInt("10", foo);"#, None),
+        (r#"parseInt("10", 10);"#, Some(json!(["always"]))),
+        (r#"parseInt("10");"#, Some(json!(["as-needed"]))),
+        (r#"parseInt("10", 8);"#, Some(json!(["as-needed"]))),
+        (r#"parseInt("10", foo);"#, Some(json!(["as-needed"]))),
+        ("parseInt", None),
+        ("Number.foo();", None),
+        ("Number[parseInt]();", None),
+        ("class C { #parseInt; foo() { Number.#parseInt(); } }", None),
+        ("class C { #parseInt; foo() { Number.#parseInt(foo); } }", None),
+        ("class C { #parseInt; foo() { Number.#parseInt(foo, 'bar'); } }", None),
+        ("class C { #parseInt; foo() { Number.#parseInt(foo, 10); } }", Some(json!(["as-needed"]))),
+        ("var parseInt; parseInt();", None),
+        ("var parseInt; parseInt(foo);", Some(json!(["always"]))),
+        ("var parseInt; parseInt(foo, 10);", Some(json!(["as-needed"]))),
+        ("var Number; Number.parseInt();", None),
+        ("var Number; Number.parseInt(foo);", Some(json!(["always"]))),
+        ("var Number; Number.parseInt(foo, 10);", Some(json!(["as-needed"]))),
+        // ("/* globals parseInt:off */ parseInt(foo);", Some(json!(["always"]))),
+        // ("Number.parseInt(foo, 10);", Some(json!(["as-needed"]))), // { globals: { Number: "off" } }
```

Comment:
Do we care about these? The only rule that kinda checks for globals is `no_undef`, but even that one has a bunch of commented out tests.

Reply:
We can leave these out for now.